### PR TITLE
Fix norm decomp when dtype is passed in

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89508

Fix for https://github.com/pytorch/torchdynamo/issues/1889. The wrapper was doing a downcast even when the dtype was explicitly passed in.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by anijain2305 — APPROVED** [review]

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -1775,17 +1775,23 @@ def log_sigmoid_forward(self: Tensor) -> Tuple[Tensor, Tensor]:
...
     dtype: Optional[torch.dtype] = None,
 ):
-    if p is None:
-        p = 2.0
-    return torch.linalg.vector_norm(self, p, dim, keepdim, dtype=dtype)
+    p = p if p is not None else 2.0
+    if dtype:
+        return torch.linalg.vector_norm(self, p, dim, keepdim, dtype=dtype)
+
+    computation_dtype, result_dtype = utils.elementwise_dtypes(
```
Comment (ngimel) [inline torch/_decomp/decompositions.py:1789]: do you need this? Just calling `vector_norm` should be enough. Alternatively, we can switch to `norm` decomposition from `_refs` (and it's just calling `torch.linalg.vector_norm`, passing through dtype arg).

Comment (eellison) [reply torch/_decomp/decompositions.py:1789]: i think it's still good to model the dtype conversion so inductor can fuse it

Comment (ngimel) [reply torch/_decomp/decompositions.py:1789]: Inductor gets fusable dtype conversions from `linalg_vector_norm` decomposition that is uses, so I really think there's no point in this decomp, we should just add registration to the one in `_refs` that does the same thing.

**Pull request closed without merging**